### PR TITLE
refactor tree into new component and code cleaning

### PR DESCRIPTION
**Pull request by konolak**

### CODE REVIEW

File: mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+  }
+
+  function searchFromTree(input: any) {
+    clearTreeSearch();
+    const hits = { nodeIds: [], childNodeIds: [] };
+    doFiltering(treeData, input.toString(), hits);
+    expandAndSelectNodes(hits.nodeIds);
+  }
+
+  function handleTreeClick(
```
Comment (maariaw) [inline mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx:201]: I have trouble understanding the need for the handleTreeClick and handleTreeToggle as intermediary functions here. Could you not put setTreeSelections(nodeIds) and setTreeExpanded(nodeIds) in their place for more concise code? And I'm curious why is the unused argument (event) typed when precise typing is missing in many other places. I'd like to learn the reasoning behind it.

File: mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    nodeIds: any,
+  ) => {
+    if (action === 'handleSelect') {
+      handleTreeClick(event, nodeIds);
+    } else if (action === 'treeToggle') {
+      handleTreeToggle(event, nodeIds);
+    }
+  };
+
+  const performNodeInfoAction = (nodeId: any, isSourceTree: boolean) => {
```
Comment (maariaw) [inline mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx:233]: Is this for some future use? NodeInfo doesn't seem to be using the performNodeInfoAction as of now. If it's not going to be in use, both this and selectFromTreeById don't have a use.

File: mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+    setTreeExpanded(nodeIds);
+  }
+
+  const selectFromTreeById = (nodeId: string) => {
+    const nodeIds = [];
+    nodeIds.push(nodeId);
+    expandAndSelectNodes(nodeIds);
+  };
+
+  const performCallbackFromTreeAction = (
```
Comment (maariaw) [inline mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx:221]: I'm no expert with Redux either, but I have a feeling using the redux store for these selected and expanded states would help get rid of these long chains of passing functions.

File: mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx (hunk 1 above)
Comment (konolak) [reply mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx:201]: These might be originated from some tree examples. Probably could be refactored into shorter code. This missing event was causing some errors/warnings at some earlier stage, so it needed to be fixed, but might be "legacy" already.

File: mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx (hunk 2 above)
Comment (konolak) [reply mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx:233]: If multiple nodes are selected from tree, there could be a possibility to select the node to be mapped from the node drop down. But this is a stub and some more discussion about details is needed.

File: mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx (hunk 3 above)
Comment (konolak) [reply mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx:221]: I agree on this. The whole state management could be simplified with using dispatchers.

**Pull request closed without merging**